source/common/http/http1/balsa_parser.cc

```diff
@@ -13,9 +13,24 @@ namespace Envoy {
 namespace Http {
 namespace Http1 {
 
+namespace {
+
 using ::quiche::BalsaFrameEnums;
 using ::quiche::BalsaHeaders;
 
+bool isMethodValid(absl::string_view method) {
+  static const absl::flat_hash_set<absl::string_view> kValidMethods{
```

Comment:
I think this may run into https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables. One way to adhere to this is via something like:

  static const absl::flat_hash_set<absl::string_view>* const kValidMethods =
      new absl::flat_hash_set<absl::string_view>{"ACL", "BIND", ...};